the-state-of-doctests-in-haskell/article.md

```diff
@@ -63,191 +63,199 @@ Follow the steps below in order to launch the doctest examples in our `doctests-
+
+2. The library doesn't depend on GHC as a library, so it's much more resilient to GHC version changes.
+
+3. The library doesn't require recompiling the sources if it's only about writing doctests - this saves additional time for developers and
+makes writing documentation pleasant.
+
 All of the above makes it more than a viable alternative.
 
 However, the library seems to be closely bound to [`cabal-install`](https://wiki.haskell.org/Cabal): it uses `cabal-install` generated metadata - the `plan.json` file.
 
 Let's consider using `cabal-docspec` to run the doctests in our `doctests-demo` project:
 
-1. First, setup `cabal-install` and the global compiler alongside with all of the auxiliary tooling.
+1. First, setup `cabal-install` and the global compiler.
 We suggest using [ghcup](https://www.haskell.org/ghcup/) to achieve this.
 
 2. Build the project using `cabal`:
 
-    `cabal v2-build`
+`cabal v2-build`
 
 3. Download `cabal-docspec` binaries from [the releases page](https://github.com/phadej/cabal-extras/releases).
 You can also consider building the library from source, following the instructions [here](https://github.com/phadej/cabal-extras).
 
-    ```
-    curl -sL https://github.com/phadej/cabal-extras/releases/download/cabal-docspec-0.0.0.20211114/cabal-docspec-0.0.0.20211114.xz > cabal-docspec.xz
-    xz -d < cabal-docspec.xz > "$HOME"/.local/bin/cabal-docspec
-    rm -f cabal-docspec.xz
-    chmod a+x "$HOME"/.local/bin/cabal-docspec
-    ```
+```
+curl -sL https://github.com/phadej/cabal-extras/releases/download/cabal-docspec-0.0.0.20211114/cabal-docspec-0.0.0.20211114.xz > cabal-docspec.xz
+xz -d < cabal-docspec.xz > "$HOME"/.local/bin/cabal-docspec
+rm -f cabal-docspec.xz
+chmod a+x "$HOME"/.local/bin/cabal-docspec
+```
 
 4. Now, run the doctest examples using `cabal-docspec`:
 
-    `cabal-docspec`
+`cabal-docspec`
 
 5. The command above should fail with the following error:
 
-    ```
-    expected: "{\"title\":\"One-Punch Man\",\"rating\":8.9}"
-    but got:
-              ^
-              <interactive>:10:1: error:
-                  Variable not in scope: encode :: Anime -> t
-    ```
+```
+expected: "{\"title\":\"One-Punch Man\",\"rating\":8.9}"
+but got:
+            ^
+            <interactive>:10:1: error:
+                Variable not in scope: encode :: Anime -> t
+```
 
 6. The above happens because the library handles things a little differently - it requires modules to be explicitly imported/exported.
 Let's edit our doctests a little to make `cabal-docspec` work:
 
-    ```haskell
-    -- src/Sample.hs
-
-    {-# LANGUAGE OverloadedStrings #-}
-    {-# LANGUAGE TemplateHaskell   #-}
-
-    module Sample where
+<details>
+<summary>Workaround</summary>
 
-    import Data.Text
-    import Data.Aeson
-    import Data.Aeson.TH
+```haskell
 
-    data Anime =
-      Anime { title  :: Text
-            , rating :: Double
-            }
+-- |
+-- >>> import Data.Aeson
```

Comment:
I still think this could be made more evident. Two solutions that would work nicely:
1. Right after the "Workaround", mention that you just added this import, or
2. Change syntax highlighting from "haskell" to "diff" and add a + on the first column of this line.